database/migrations/2025_06_23_161358_add_tool_id_column_to_documents_table.php

```diff
@@ -0,0 +1,25 @@
+<?php
+
+use Illuminate\Database\Migrations\Migration;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Support\Facades\Schema;
+
+return new class extends Migration
+{
+    public function up(): void
+    {
+        Schema::table('documents', function (Blueprint $table) {
+            $table->foreignId('tool_id')
+                ->nullable()
+                ->constrained()
+                ->onDelete('cascade');
```

Comment:
Deleting the tool will delete the associated documents. I'm not sure that this is correct. However, if this is the correct behaviour, there is another issue where deleting the tool deletes the document, but the underlying revisions aren't deleted.